runtime/gc_glue_java/ArrayletObjectModelBase.hpp

```diff
@@ -252,7 +252,7 @@ class GC_ArrayletObjectModelBase
 			 * of the after-last element without crashing. Handle the case of UDATA_MAX specially, since we use that
 			 * for any object whose size overflows the address space.
 			 */
-			UDATA dataSizeInBytes = (UDATA_MAX == unadjustedDataSizeInBytes) ? UDATA_MAX : (unadjustedDataSizeInBytes + 1);
+			UDATA dataSizeInBytes = (UDATA_MAX == unadjustedDataSizeInBytes) ? UDATA_MAX : unadjustedDataSizeInBytes;
```

Comment:
There is no adjustment any more. This line and comment above should go away. The formal parameter should be renamed unadjustedDataSizeInBytes->dataSizeInBytes .